src/sentry/api/endpoints/organization_stats_v2.py

```diff
@@ -0,0 +1,239 @@
+            stat_to_update.update(row)
+
+        # response.zerofill(start, end, rollup)
+        return Response(response.build_fields())
+
+
+class OrganizationProjectStatsIndex(OrganizationEndpoint):
+    def get(self, request, organization):
+        start, end, rollup = get_date_range_rollup_from_params(request.GET, "1h", round_range=True)
+
+        # TODO: see if there's a better way to get the user's projects
+        project_list = []
+        team_list = Team.objects.get_for_user(organization=organization, user=request.user)
+        for team in team_list:
+            project_list.extend(Project.objects.get_for_user(team=team, user=request.user))
+        project_ids = list({p.id for p in project_list})
+
+        result = outcomes.query(
+            start=start,
+            end=end,
+            rollup=rollup,
+            groupby=["project_id", "category", "time", "outcome", "reason"],
+            aggregations=[["sum", "times_seen", "times_seen"], ["sum", "quantity", "quantity"]],
+            filter_keys={"org_id": [organization.id], "project_id": project_ids},
+            orderby=["times_seen", "time"],
+        )
+
+        response = {project_id: StatsResponse(start, end, rollup) for project_id in project_ids}
+        for row in result:
+            stat_to_update = response[row["project_id"]].get(row["category"])
+            stat_to_update.update(row)
+
+        return Response(
+            {project_id: stat_res.build_fields() for project_id, stat_res in response.items()}
+        )
+
+
+class OrganizationProjectStatsDetails(ProjectEndpoint, OrganizationEndpoint):
+    def get(self, request, project):
+        start, end, rollup = get_date_range_rollup_from_params(request.GET, "1h", round_range=True)
+
+        # TODO: see if there's a better way to get the user's projects
+        project_list = []
+        team_list = Team.objects.get_for_user(organization=project.organization, user=request.user)
+        for team in team_list:
+            project_list.extend(Project.objects.get_for_user(team=team, user=request.user))
+        project_ids = list({p.id for p in project_list})
+
+        if project.id not in project_ids:
+            # TODO: more info in response here?
+            return Response(status=404)
+
+        result = outcomes.query(
+            start=start,
+            end=end,
+            rollup=rollup,
+            groupby=["category", "time", "outcome", "reason"],
+            aggregations=[["sum", "times_seen", "times_seen"], ["sum", "quantity", "quantity"]],
+            filter_keys={"org_id": [project.organization.id], "project_id": [project.id]},
+            orderby=["time"],
+        )
+        response = StatsResponse(start, end, rollup)
+        for row in result:
+            stat_to_update = response.get(row["category"])
+            stat_to_update.update(row)
+
+        return Response(response.build_fields())
+
+
+# TODO: good names for these classes
+class StatsResponse:
```

Comment:
Other parts of the application have this kind of logic in `sentry.api.serializers`. There are also some useful base classes in `sentry.api.serializers.snuba` that help with zerofilling timeseries results.

Reply:
would it be okay for these classes to live in `serializers/models` or do we limit those to just ones with corresponding django serializers? i see in a few other places we create a top level folder for endpoints w/ helper classes included in the folder

Reply:
I'm fine with them being in serializers, as that directory also contains DRF serializers for request parsing. The top level directory for endpoints was something we tried where parts of the product could be separate django applications, but it didn't really get that much traction.